csrc/tensor_view.cpp

```diff
@@ -1229,8 +1232,33 @@ TensorView* TensorView::cacheBefore(LoadStoreOpType cache_op) {
 
   auto replayed_consumer_pair =
       TransformReplay::replayCasP(consumer, producer, -1);
+
   consumer->setDomain(replayed_consumer_pair.first);
 
+  // Recover allocation domain from transform replay
+  // TODO: should move this to
```

Comment:
The following code looks good to me, but I'm not sure what you're saying with this comment.

Reply:
Updated the comment. Basically the logic below is just equivalent to asking `replayCasP` to replay the allocation domain of the consumer consistent with the producer. So we should consider making it an opt-in functionality of `replayCasP` if there are other use cases needing this.

Reply:
So, `replayCasP` also set its allocation domain as the producer transformations are replayed on the consumer? Having that option seems to make sense.